src/charm.py

```diff
@@ -222,6 +229,33 @@ def issue_certificate(self, certificate_data, cert_type):
         logging.info(f"new cert data: {response['data']}")
         return response['data']
 
+    def _on_generate_certificate_action(self, event) -> None:
+        """Generates TLS Certificate.
+
+        Generates a private key and certificate for an external service.
+        Args:
+            event: Juju event.
+        Returns:
+            None
+        """
+        data = {
+            "common_name": event.params["cn"],
+            "certificate_name": event.params["cn"],
+            "sans": event.params["sans"],
+        }
```

Comment:
Please see the previous comment, something like this can resolve 

        sans_str = event.params["sans"]
        sans = json.dumps(sans_str.split(" "))
        data = {
            "common_name": event.params["cn"],
            "certificate_name": event.params["cn"],
            "sans": sans,
        }

Reply:
Good catch - I've pushed a commit something like the above to deal with this